lib/types/protocol.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package common
+package types
 
 type Protocol struct {
```

Comment:
Yeah, good point.

Though in that case I think I would still question the use of net.IP in the `api` structs at all (but that is a separate issue from this).

My original concern was to make sure package names were based more around specific sets of function.  A package named `types` has strong potential to grow into something that's big and ugly, and not well defined, whereas a package named `net` has much more clearly defined boundaries. This also matches the go package naming conventions.

i.e keeping a `types` package sort of defeats a lot of the purpose of splitting up the old `common` package.